test/test_audio.rb

```diff
@@ -114,4 +114,26 @@ def test_song
     
     assert_nil Gosu::Song.current_song
   end
+
+  def test_song_finished_callback
+    skip_on_appveyor
+
+    win = SongTestWindow.new
+    assert_nil Gosu::Song.current_song
+
+    song1 = Gosu::Song.new(media_path("0614.ogg"))
+    song2 = Gosu::Song.new(media_path("0830.ogg"))
+
+    song1.play(false) { song2.play }
+    assert_equal song1, Gosu::Song.current_song
+
+    assert song1.playing?
+    refute song2.playing?
+
+    (20 * 1000/win.update_interval).ceil.times{ win.tick }
```

Comment:
Maybe I should look for a shorter first sound-clip as 20s are quite some time...